### PR TITLE
added unit tests for extractMetadata helper functions

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
Added unit test for helper function in `extractMetadata.ts`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -10,22 +10,22 @@ overrides:
 dependencies:
   '@casl/ability':
     specifier: ^6.3.3
-    version: 6.7.0
```
Comment (saw-jan) [inline pnpm-lock.yaml:13]: not expecting the change in lock file

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/unit/helper/extractMetadata.spec.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  formatTimeTagChecker,
+  addSopTagChecker,
+  formatDate,
+  formatTime,
+  fetchDicomImageData,
+  extractDicomMetadata
+} from '../../../src/helper/extractMetadata'
+import dicomTags from '../../../src/helper/dicomTags'
+
+const imageData = {
```
Comment (saw-jan) [inline tests/unit/helper/extractMetadata.spec.ts:14]:
```suggestion
const mockImageData = {
```

File: tests/unit/helper/extractMetadata.spec.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    it('should return extracted dicom metadata', async () => {
+      const imageData = {
+        string: function (tag: string) {
+          switch (tag) {
+            case 'x00100010':
+              return 'testPatient'
+          }
+        }
+      }
+      const dicomMetaData = await extractDicomMetadata(imageData, ['patientName'], 'en')
```
Comment (Salipa-Gurung) [inline tests/unit/helper/extractMetadata.spec.ts:160]: maybe we can add another test case where the tags has to be removed.

File: tests/unit/helper/extractMetadata.spec.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              return 'testPatient'
+          }
+        }
+      }
+
+      const dicomMetaData = await extractMetadata.extractDicomMetadata(
+        imageData,
+        ['patientName', '', undefined, null],
+        'en'
+      )
```
Comment (koebel) [inline tests/unit/helper/extractMetadata.spec.ts:42]: I think you need to test more inputs. In addition to a standard tag like "patientName" that just returns the value from meta data and empty/undefined value you need to add
- some tag with _formatDate like for example "patientBirthdate" which should return a date value that has been formatted nicely
- some tag with _formatTime like instanceCreationTime which should return a time value that has been formatted nicely
- some tag with _addSOPuids like SOP_ClassUID which should return the SOP class together with the description
You might also want to test what happend if someone passes an invalid tag into this function (I think it should return undefined)

File: src/helper/extractMetadata.ts [hunk 5]
```diff
@@ -65,6 +65,7 @@ export const extractDicomMetadata = async (imageData: object, tags: string[], la
       value: metadataValue
     })
   }
+  console.log(extractedData)
```
Comment (koebel) [inline src/helper/extractMetadata.ts:68]: I guess this was just for "spying" on the values produced by this function :)

File: src/helper/extractMetadata.ts [hunk 6]
```diff
@@ -73,6 +74,7 @@ const formatDate = (date: string, language: string, dateFormat: DateTime): strin
   // transforming date into a string that is valid for formatDateFromISO ('YYYY-MM-DDTHH:MM:SS')
   // description of input format see https://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html, VR Name 'DA'
   // date output format: e.g. DateTime.DATE_MED, DateTime.DATE_SHORT, see https://moment.github.io/luxon/api-docs/index.html
+  console.log(date)
```
Comment (koebel) [inline src/helper/extractMetadata.ts:77]: same as above...

File: src/helper/extractMetadata.ts (hunk 5 above)
Comment (koebel) [reply src/helper/extractMetadata.ts]: no probs

**Review by koebel — APPROVED** [review]
LGTM - however get some feedback from others in the team too before merging

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**